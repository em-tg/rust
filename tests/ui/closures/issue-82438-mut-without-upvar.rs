//@ check-pass
use std::error::Error;
struct A {
}

impl A {
    pub fn new() -> A {
        A {
        }
    }

    pub fn f<'a>(
        &'a self,
        team_name: &'a str,
        c: &'a mut dyn FnMut(String, String, u64, u64)
    ) -> Result<(), Box<dyn Error>> {
        Ok(())
    }
}


fn main() {
    let a = A::new();
    let participant_name = "A";

    let c = |a, b, c, d| {};

    a.f(participant_name, &mut c); //~ WARNING cannot borrow
}
