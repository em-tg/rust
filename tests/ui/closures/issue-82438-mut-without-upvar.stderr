warning[E0596]: cannot borrow `c` as mutable, as it is not declared as mutable
  --> $DIR/issue-82438-mut-without-upvar.rs:28:27
   |
LL |     a.f(participant_name, &mut c);
   |                           ^^^^^^ cannot borrow as mutable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider changing this to be mutable
   |
LL |     let mut c = |a, b, c, d| {};
   |         +++

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0596`.
